sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/api/XrayAPIResponseFactory.java

```diff
@@ -0,0 +1,87 @@
+     * "https://github.com/eugenp/tutorials/blob/master/core-java-modules/core-java-networking-2/src/main/java/com/baeldung/httprequest/FullResponseBuilder.java"/a>
+     *
+     * @param httpURLConnection httpUrl connection
+     * @param zipArchive        file name to save zip content
+     * @return xray api http response
+     * @throws XrayWrapperException
+     */
+    public XrayAPIResponse createHttpResponseFromConnection(HttpURLConnection httpURLConnection, String zipArchive) throws XrayWrapperException {
+
+        // read response code
+        int responseCode = 0;
+        try {
+            responseCode = httpURLConnection.getResponseCode();
+        } catch (IOException e) {
+            throw new XrayWrapperException("Could not get response code from HTTP connection.", XrayWrapperExitCode.IO_ERROR, e);
+        }
+
+        // red response Message
+        String responseMessage;
+        try {
+            responseMessage = httpURLConnection.getResponseMessage();
+        } catch (IOException e) {
+            throw new XrayWrapperException("Could not read Response Message from HTTP connection.", XrayWrapperExitCode.IO_ERROR, e);
+        }
+
+        // read headers
+        Map<String, List<String>> header = httpURLConnection.getHeaderFields();
+
+        // get input stream from response, client accept status codes 3xx
+        if (responseCode > 399) {
+            try (InputStream inputStream = httpURLConnection.getErrorStream()) {
+                if (inputStream != null) {
+                    String content = saveOrReadHTTPContent(httpURLConnection, zipArchive, inputStream);
+                    return XrayAPIResponse.Builder.builder().statusCode(responseCode).headers(header).addResponseBody(content)
+                            .addResponseMessage(responseMessage).build();
+                }
+            } catch (IOException e) {
+                throw new XrayWrapperException("Could not save https input stream to zip file", XrayWrapperExitCode.IO_ERROR, e);
+            }
+
+        } else {
+            try (InputStream inputStream = httpURLConnection.getInputStream()) {
+                String content = saveOrReadHTTPContent(httpURLConnection, zipArchive, inputStream);
+                return XrayAPIResponse.Builder.builder().statusCode(responseCode).headers(header).addResponseBody(content).addResponseMessage(responseMessage)
+                        .build();
+            } catch (IOException e) {
+                throw new XrayWrapperException("Could not save https input stream to zip file", XrayWrapperExitCode.IO_ERROR, e);
+            }
+        }
+
+        return XrayAPIResponse.Builder.builder().statusCode(responseCode).headers(header).addResponseMessage(responseMessage).build();
+    }
+
+    private String saveOrReadHTTPContent(HttpURLConnection httpURLConnection, String zipArchive, InputStream inputStream) throws XrayWrapperException {
```

Comment:
I am sorry that I did not see this before, but here we have method with logic inside where the caller has no clue if there shall be string returned or not.

Would it not make more sense to have two methods: "saveHTTPContent" and "readHTTPContentAsString"? 
At line 81 always an empty string is returned - which makes no sense for callers to handle it in any way. I think IMHO that it should be clear at caller side if there is need of storage or a need to fetch a string